documentation-website/Writerside/topics/DSL-CRUD-operations.topic

```diff
@@ -376,65 +328,35 @@
             order, provide a list of columns as the second parameter:</p>
-
-        <code-block lang="kotlin">
-            val userCount = users.selectAll().count()
-            users.insert(
-                users.select(
-                    stringParam(&quot;Foo&quot;),
-                    Random().castTo&lt;String&gt;(VarCharColumnType()).substring(1, 10)
-                ), columns = listOf(users.name, users.id)
-            )
-        </code-block>
+        <code-block lang="kotlin"
+                    src="exposed-dsl/src/main/kotlin/org/example/examples/InsertSelectExamples.kt"
+                    include-symbol="userCount, insertedUsers"
+        />
     </chapter>
     <chapter title="Replace" id="replace">
         <tldr>
             <p>Supported on: SQLite, MySQL, and MariaDB </p>
         </tldr>
         <p>
-            The <code>replace()</code>
+            The
+            <a href="https://jetbrains.github.io/Exposed/api/exposed-core/org.jetbrains.exposed.sql/replace.html">
+                <code>replace()</code>
+            </a>
             method acts in a similar manner to an <code>upsert()</code>. The only difference is that
             if an insertion would violate a unique constraint, the existing row is deleted before the new row is inserted.
         </p>
 
-        <code-block lang="kotlin">
-            object StarWarsFilms : Table() {
-                val sequelId: Column&lt;Int&gt; = integer(&quot;sequel_id&quot;).uniqueIndex()
-                val releaseYear: Column&lt;Int&gt; = integer(&quot;release_year&quot;)
-                val name: Column&lt;String&gt; = varchar(&quot;name&quot;, 50)
-                val director: Column&lt;String&gt; = varchar(&quot;director&quot;, 50)
-                val rating: Column&lt;Double&gt; = double(&quot;rating&quot;).default(10.0)
-
-                override val primaryKey = PrimaryKey(sequelId, releaseYear)
-            }
-
-            transaction {
-                // ...
-                // inserts a new row with default rating
-                StarWarsFilms.replace {
-                    it[sequelId] = 9
-                    it[releaseYear] = 2019
-                    it[name] = &quot;The Rise of Skywalker&quot;
-                    it[director] = &quot;JJ Abrams&quot;
-                }
-                // deletes existing row and inserts new row with set [rating]
-                StarWarsFilms.replace {
-                    it[sequelId] = 9
-                    it[releaseYear] = 2019
-                    it[name] = &quot;The Rise of Skywalker&quot;
-                    it[director] = &quot;JJ Abrams&quot;
-                    it[rating] = 5.2
-                }
-            }
-        </code-block>
+        <code-block lang="kotlin"
+                    src="exposed-dsl/src/main/kotlin/org/example/examples/ReplaceExamples.kt"
+                    include-symbol="replace"
+        />
```

Comment:
**Minor:** This is just an opinion and I don't think it needs to change in this PR (or ever actually).
I think it sometimes is distracting to have the relevant sample code wrapped in a function block. Especially since it's only for some code blocks but not for all.
I don't, however, think it will bring less value to the users or adversely affect them. I'm just bringing it up because I definitely felt a difference every time reading a code block that went straight to the point versus a block where I had to step into a function first.
I do appreciate that it's easier to include a symbol than to use specific lines and it means easier maintenance and less room for future errors. So overall I see the advantage of keeping it this way and am good with it.

Reply:
I agree that the wrapping function is not necessary, especially as the reader doesn't have the context of the project. I already adjusted some of the references. We can still use symbols where it makes sense and for the rest we'll just have to remember to update the `code-block`s, but at least the examples will look clean. ✌️